lib/active_model/serializer/reflection.rb

```diff
@@ -47,11 +47,23 @@ class Serializer
     #
     # So you can inspect reflections in your Adapters.
     class Reflection < Field
+      attr_reader :foreign_key, :type
+
       def initialize(*)
         super
         options[:links] = {}
         options[:include_data_setting] = Serializer.config.include_data_default
         options[:meta] = nil
+        @type = options.fetch(:type) do
+          class_name = options.fetch(:class_name, name.to_s.camelize.singularize)
```

Comment:
This line is causing the bug reported here: https://github.com/rails-api/active_model_serializers/issues/2160 In particular, the use of the field's `name` if the `class_name` and `type` aren't specified breaks belongs_to associations.